kaggle_environments/envs/lux_ai_2021/lux_ai_2021.py

```diff
@@ -140,15 +164,32 @@ def interpreter(state, env):
         player2.status = "DONE"
     return state
 
+def get_message(dimension_process):
+    raw = dimension_process.stderr.readline()
+    try:
+        res = json.loads(raw)
+        return res
+    except Exception as e:
+        print("Engine Exception")
+        err_stack = dimension_process.stderr.readlines(100)
+        # err_stack = [raw, *err_stack]
+        # print(err_stack)
+        for m in err_stack:
+            if len(m) < 1000: 
+                print(m.decode(), file=sys.stderr)
+            else:
+                print("...", file=sys.stderr)
+
 def filter_actions(state, env):
     enable_annotations = env.configuration["annotations"]
     if not enable_annotations:
         for team in range(len(state)):
             filtered = []
-            for l in state[team].action:
-                if len(l) > 0 and l[0] != "d":
-                    filtered.append(l)
-            state[team].action = filtered
+            if state[team] is not None and state[team].action is not None:
```

Comment:
this fixes the bug where validation episodes that failed (or any episode that fails actually) do not show up at all in the competition

Reply:
this might also be why the competition servers are slow? Since a bunch of matches aren't running due to either timeouts, errors etc.

Reply:
👍 engine should be able to return a descriptive log message if either suggestion is missing a main.py in the top level directory.